File: vis-viva/app/src/model/battle-units/classes/FireWorm.js
import BattleUnit from "../BattleUnit.js";
import L from "../../utils/Language.js";
const res = require.context('../../../../res/battle-units/fire-worm', false, /\.(png|jpe?g|svg|mp3|gif)$/)

export default class FireWorm  extends BattleUnit{

    static TYPE = "Fire Worm"
    static LIST_RANK = 2


    constructor(){
        super(FireWorm.TYPE, 100, 15)
        this.importResources(res)
        this.__attack_duration = 1600
        this.__die_duration = 1500
        this.__take_hit_duration = 600
        this.missRate = 0.3
        this.criticalHitRate = 0.8
        this.criticalHitMultiplier = 3
        this.dodgeRate = 0.1
        this.description = L.fire_worm_description
    }

}

File: server/battle_units/FireWorm.py
from battle_unit import BattleUnit

class FireWorm(BattleUnit):
    
    def __init__(self, faction, position):
        BattleUnit.__init__(self, faction, position, "Fire Worm", 100 , 100)
        self["missRate"] = 0.3
        self["criticalHitRate"] = 0.8
        self["dodgeRate"] = 0.1
        self["criticalHitMultiplier"] = 3
        self["damage"] = 15